### PR TITLE
Support half page scrolling using Emacs key bindings.

### PR DESCRIPTION
**Pull request by amosbird**

### CODE REVIEW

**Comment by junegunn** [comment]
Thanks for the suggestion. Are ctrl-v and alt-v standard bindings in Emacs? If it's not the case, I don't want to provide those bindings by default. Instead, we add those actions, and let the users set up their own bindings using `--bind`.

**Comment by amosbird** [comment]
It's the default binding in emacs.

**Comment by junegunn** [comment]
Half-page or whole-page? If those keys are for whole-page movement in Emacs, there's no reason we should bind them differently in fzf.

**Comment by amosbird** [comment]
Oh, I remapped them a long time ago and it becomes my builtin default 😆 . My bad.

**Comment by junegunn** [comment]
I see, then we can provide unbound `half-page-up` and `half-page-down` actions and let the users bind those actions to the keys of their choices via `$FZF_DEFAULT_OPTS`.

File: man/man1/fzf.1 [hunk 1]
```diff
@@ -444,6 +444,8 @@ e.g. \fBfzf --bind=ctrl-j:accept,ctrl-k:kill-line\fR
     \fBnext-history\fR          (\fIctrl-n\fR on \fB--history\fR)
     \fBpage-down\fR             \fIpgdn\fR
     \fBpage-up\fR               \fIpgup\fR
+    \fBhalfpage-down\fR
+    \fBhalfpage-up\fR
```
Comment (junegunn) [inline man/man1/fzf.1]: Please use `half-page` instead of `halfpage`.

File: src/options.go [hunk 2]
```diff
@@ -667,6 +667,10 @@ func parseKeymap(keymap map[int]actionType, execmap map[int]string, str string)
 			keymap[key] = actPageUp
 		case "page-down":
 			keymap[key] = actPageDown
+		case "halfpage-up":
```
Comment (junegunn) [inline src/options.go]: Same here.

File: src/terminal.go [hunk 3]
```diff
@@ -1413,6 +1415,12 @@ func (t *Terminal) Loop() {
 			case actPageDown:
 				t.vmove(-(t.maxItems() - 1))
 				req(reqList)
+			case actHalfPageUp:
+				t.vmove(t.maxItems()/2 - 1)
+				req(reqList)
+			case actHalfPageDown:
+				t.vmove(-(t.maxItems()/2 - 1))
```
Comment (junegunn) [inline src/terminal.go]: `- 1` doesn't seem necessary here. The point of `t.maxItems() - 1` is to keep the scroll offset unchanged when you press pgup for the first time. It doesn't apply for half page movements.

**Comment by amosbird** [comment]
updated.

**Comment by junegunn** [comment]
Oops, I forgot to include this in the new release. I'll make sure it's included in the next version. Sorry for the delay.

**Comment by junegunn** [comment]
Released 0.16.1 with your path. Thanks.

**Pull request merged**